### ISSUE TITLE
Fix exec::structures::get_field_and_mutate::compiled and exec::change_field::change_field::compiled

### ISSUE THREAD

**Issue by jarctan** [jarctan/vache#17]
What does not work:
- exec::structures::get_field_and_mutate::compiled and exec::change_field::change_field::compiled

Why:
```
error[E0502]: cannot borrow `john` as mutable because it is also borrowed as immutable
      --> src/main.rs:787:71
       |
   785 |             let mut n: Var<__String> = __ref(&(john).name);
       |                                               ------ immutable borrow occurs here
   786 |             *(n) = Cow::owned(__String::from("12"));
   787 |             println!("{} {}", Var::__take(& mut n), Var::__take(& mut (john).name));
       |             ----------------------------------------------------------^^^^^^-------
       |             |                                                         |
       |             |                                                         mutable borrow occurs here
       |             immutable borrow later used here
```

**Comment by jarctan** [comment]
#15

**Comment by jarctan** [comment]
For change_field, the real issue is with this line:
```
*(n) = Cow::owned(__String::from("12"));
```
Where we know that we are entirely replacing n, so n should not be linked to any previous borrow whatsoever. In particular, not to `john.name`.
But because we dereference `n`, rust cannot infer that n is entirely replaced here. So the borrow on `john.name` lives on in the view of the compiler, until the line:
```
println!("{} {}", Var::__take(& mut n), Var::__take(& mut (john).name));
```
Where it sees that we are moving `john.name`, but `n` still references `john.name` in its view, so we cannot move john.name out.

**Comment by jarctan** [comment]
Seemingly fixed by sledgehammer commit 2fd11bdc716c457b3175026179577d265d3e37e7
